application/models/services/SurveyPermissions.php

```diff
@@ -173,6 +182,14 @@ public function addUserGroupToSurveyPermissions($userGroupId)
      */
     public function getSurveyUserList()
     {
+        $hasUserReadPermission = \Permission::model()->hasGlobalPermission('users', 'read');
+
+        // If the user doesn't have User Read permission, and it's not allowed
+        // to see at least the users of the same group, then return an empty list
+        if (!$hasUserReadPermission && !$this->userControlSameGroupPolicy) {
```

Comment:
It's not the proposed solution on https://bugs.limesurvey.org/view.php?id=18468#c75188

It must show all users (like in current 3.X)

And see : https://bugs.limesurvey.org/view.php?id=18468#c75192

Reply:
> It must show all users (like in current 3.X)

Actually, that's the part we said it had to be corrected.
Right now it can see al users. But if no permission, it should not see any user.
Row 6 and 8 of the table in the ticket
![image](https://github.com/LimeSurvey/LimeSurvey/assets/4942495/2b879a25-fb37-4a1a-874f-43d4cbb55e8b)

Reply:
1. In 3. X : you can allow any user to give permission to any other user without access to User list (with email and login).
It's not possible after this fix.

2. In 3.X : you can allow an user to see only some user in User Management : it's not possible after this update.

Reply:
Can we pick this up as to wrap this PR?
Can you please describe a bit more the situations 1?

As per number 2, I believe we are not updating anything from user managemnt screen.

Reply:
@Shnoulle Ping for reminder :)

Reply:
> @Shnoulle Ping for reminder :)

I report issue and feature : 
- https://bugs.limesurvey.org/view.php?id=18921
- https://bugs.limesurvey.org/view.php?id=18916

My global opinion :  we need more detail on needed behavior …

I check in 3.X (and 5.X): an user without Permission on User see ALL user in Survey permission (no group management used)

Reply:
We need a list with all option possible and for each what is the desired behavior and what is the current.

If current was updated : it's a major API version update : can only be in LimeSurvey 7 (in my opinion)

Reply:
> We need a list with all option possible and for each what is the desired behavior and what is the current

We did this already. right?
https://bugs.limesurvey.org/view.php?id=18468#c75188

That's where we detected the bug.

Reply:
Yes, but there are 2 issues with this list

1. We don't separate Seeing user in user management and in List : did we need to do it or not ? Maybe have a settings in security, at same place than _Group member can only see own group:_  something like _User need user permission to see user in list_ (current situation : No) : apply to line with _Empty list_ in the table
2. Where you set Expected to _All users_ : currently this user didn't see anybody, after update they see all user: **It can be a big issue for some LimeSurvey user** , maybe you show a client to another client.
3. The Expected to _All users_ is totally with this commit : https://github.com/LimeSurvey/LimeSurvey/commit/4824bc93f08ac4e504a2464ecf0616fffe38a0f5 where action is done ONLY if user is parent_id or have superadmin permission


The point 3 is really important here. If you set Permission View all : you see ALL user, then if you set update ALL : it seems logical to have the right to modify ALL users

And sorry : at start of this issue : i don't know for the point 3 …

Reply:
> We don't separate Seeing user in user management and in List

As far as I know the tables only concerns "giving permission to users within a survey".
I understand as well the impact of this PR only affects that process as well.

>  Maybe have a settings in security ...

I suggest to first agree on the scope of this PR.
As I understand, this is only about "giving permission to users within a survey".
If so, that setting still holds? needed? 

> Where you set Expected to All users : currently this user didn't see anybody, after update they see all user

That was part of the bug identified. If he has permission to see "all users", isn't it expected to see all users in the dropdown?

> The Expected to All users is totally at odds with this commit

That commit is about rules for delete user. This is for adding usres to a survey.
Not getting how they relate.

> The point 3 is really important here. If you set Permission View all : you see ALL user, then if you set update ALL : it seems logical to have the right to modify ALL users

Is the `update ALL` permission used here?

----

I would need someone else view here. Maybe I am missign something

Reply:
> As far as I know the tables only concerns "giving permission to users within a survey".

Yes : and then i currently have a lot of big system where _Group member can only see own group:_ is OFF to allow user to give permission to any user.

And at contrario i have client where we activate _Group member can only see own group:_ to disallow this. My client create group for each of this client.


> That was part of the bug identified. If he has permission to see "all users", isn't it expected to see all users in the dropdown?

Yes, but it's **not** the case today. I don't have this situation, but if a LimeSurvey user have this situation : you show something not shown before.
I don't know such situation, but it can exist.

> That commit is about rules for delete user. This is for adding usres to a survey.
> Not getting how they relate.

With current situation :

- Permission see user : see ALL user in all condition
- Permission delete user : delete SOME user
- Permission update user : update SOME user

I think what is expected must be

- Permission see user : see ALL user in all condition
- Permission delete user : delete ALL user
- Permission update user : update ALL user

OR 

- Permission see user : see SOME user in all condition
- Permission delete user : delete SOME user (same the  see)
- Permission update user : update SOME user (same the  see)

Reply:
As per the first 2 items you mentioned, I will let someone else to jump in and comment.

As per the last item, I believe that would be a different ticket and when tackled, we should update this dropdown list.
It is more about incongruencies in access in between list, delete and update permissions.
Not specifically related to the scope of this ticket.

Reply:
@tiborpacalat @olleharstedt @c-schmitz thoughts?